templates/common/gcp/units/gcp-update-dns.service.yaml

```diff
@@ -0,0 +1,19 @@
+name: gcp-update-dns.service
+enabled: {{if and (eq .Infra.Status.PlatformStatus.Type "GCP") .Infra.Status.PlatformStatus.GCP .Infra.Status.PlatformStatus.GCP.CloudLoadBalancerConfig (eq .Infra.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.DNSType "ClusterHosted") }}true{{else}}false{{end}}
```

Comment:
I'm not an expert of golang templating, but I think you may need each subcondition in parenthesis for the `and` to evaluate properly? 

If this isn't evaluating correctly, maybe it's easier to set it up like a function in here: https://github.com/openshift/machine-config-operator/blob/3acda20374986a357b25ef996d504914dcb0ebda/pkg/controller/template/render.go#L329

and have it just return a bool, so you can then also log to test if the condition is actually being evaluated correctly